### DISCUSSION TITLE
Unable to build on Linux Polysharp requires .NET 6.0.400 or greater

### DISCUSSION THREAD

**Discussion by DanielPower** [ppy/osu#24195] [Q&A] [answered]
I'm reproducing the same build errors on two systems. One with Fedora 38, and another with Archlinux.

**Build error**
```
PolySharp.targets(45, 5): The PolySharp source generators have been disabled on the current configuration, 
as they need Roslyn 4.3 in order to work. PolySharp requires the source generators to run in order to process 
shaders, so the library cannot be used without a more up to date IDE (eg. VS 2022 17.3 or greater) or .NET 
SDK version (.NET 6.0.400 SDK or greater).
```

IDE: JetBrains Rider with `osu.Desktop.slnf` solution
dotnet-runtime: 6.0.19
dotnet-sdk: 6.0.119

The error message indicates that `.NET 6.0.400 SDK` or greater is required. However, the latest version of dotnet core appears to be 6.0.120. 

I apologize if this is something obvious. I am not very familiar with the dotnet ecosystem. 

What I've tried so far:

- Searching the PolySharp issue tracker, but did not find anything helpful.
- Running NuGet Restore.
  - When run from Rider, it shows that all packages are already restored
  - When run from the CLI, I receive an error related to android as the target platform. 
    - I assume this is related to the README section noting a difference between Xamarin and .NET Core. Though I don't know how to specify a csproj file in the `dotnet restore` commend.

**dotnet restore error**
```
/usr/share/dotnet/sdk/6.0.119/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(112,5): error NETSDK1139: The target platform identifier android was not recognized. [/home/daniel/Development/osu/osu.Game.Rulesets.Taiko.Tests.Android/osu.Game.Rulesets.Taiko.Tests.Android.csproj]
/usr/share/dotnet/sdk/6.0.119/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(112,5): error NETSDK1139: The target platform identifier android was not recognized. [/home/daniel/Development/osu/osu.Game.Rulesets.Mania.Tests.Android/osu.Game.Rulesets.Mania.Tests.Android.csproj]
/usr/share/dotnet/sdk/6.0.119/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(112,5): error NETSDK1139: The target platform identifier android was not recognized. [/home/daniel/Development/osu/osu.Android/osu.Android.csproj]
/usr/share/dotnet/sdk/6.0.119/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(112,5): error NETSDK1139: The target platform identifier android was not recognized. [/home/daniel/Development/osu/osu.Game.Tests.Android/osu.Game.Tests.Android.csproj]
/usr/share/dotnet/sdk/6.0.119/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(112,5): error NETSDK1139: The target platform identifier android was not recognized. [/home/daniel/Development/osu/osu.Game.Rulesets.Catch.Tests.Android/osu.Game.Rulesets.Catch.Tests.Android.csproj]
/usr/share/dotnet/sdk/6.0.119/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(112,5): error NETSDK1139: The target platform identifier android was not recognized. [/home/daniel/Development/osu/osu.Game.Rulesets.Osu.Tests.Android/osu.Game.Rulesets.Osu.Tests.Android.csproj]
```

Any help to get started would be appreciated. I look forward to contributing :)

**Comment by peppy** [comment, marked as answer]
>  However, the latest version of dotnet core appears to be 6.0.120.

Please download from this page: https://dotnet.microsoft.com/en-us/download/dotnet/6.0 (latest 6.0.412).

**Reply by DanielPower** [reply]
Thank you peppy. Both Fedora and Archlinux appear to be targeting an older version. Using dotnet-install script I was able to install the latest version and osu! now builds successfully :)
